core/txpool/legacypool/list.go

```diff
@@ -392,7 +378,6 @@ func (l *list) Filter(costLimit *uint256.Int, gasLimit uint64) (types.Transactio
 	// Reset total cost
 	l.subTotalCost(removed)
 	l.subTotalCost(invalids)
-	l.txs.reheap()
```

Comment:
I think it's a very good example for consecutive filterings.

You can change `Filter` at line 374 to `filter`, and keep the reheap here.